src/ol/render/canvas/Executor.js

```diff
@@ -957,7 +957,8 @@ class Executor {
               Math.abs(textScale[0]),
               measureAndCacheTextWidth,
               font,
-              cachedWidths
+              cachedWidths,
+              viewRotationFromTransform ? 0 : this.viewRotation_
```

Comment:
Out of curiosity, in which situation does `viewRotationFromTransform` happen to be non zero?

Reply:
in the case of layer renderers that rotate the executor groups, e.g. the vector layer renderer. As opposed to the vector tile layer renderer, which renders executor groups upright.